db.go

```diff
@@ -661,17 +688,27 @@ func (db *DB) Get(dest interface{}, keys ...interface{}) error {
 // Returns an error if an element in the list has not been registered
 // with BindModel.
 func (db *DB) Insert(list ...interface{}) error {
-	return insertObjects(db, db, getInsert, list)
+	return insertObjects(db.Context(), db, db, getInsert, list)
+}
+
+// InsertContext is the context version of Insert.
+func (db *DB) InsertContext(ctx context.Context, list ...interface{}) error {
+	return insertObjects(ctx, db, db, getInsert, list)
 }
 
 // Query executes a query that returns rows, typically a SELECT. The
 // args are for any placeholder parameters in the query. This is a
 // small wrapper around sql.DB.Query that returns a *squalor.Rows
 // instead.
 func (db *DB) Query(q interface{}, args ...interface{}) (*Rows, error) {
+	return db.QueryContext(db.Context(), q, args...)
+}
+
+// QueryContext is the context version of Query.
+func (db *DB) QueryContext(ctx context.Context, q interface{}, args ...interface{}) (*Rows, error) {
 	start := time.Now()
-	if db.context != nil {
-		if deadline, ok := db.context.Deadline(); ok && !deadline.IsZero() {
+	if !isEmptyContext(ctx) {
```

Comment:
I believe the built-in support for context deadlines only applies to the driver-side connection pool. I think it's still useful to keep this for canceling queries on the MySQL server side, since closing a connection doesn't actually stop the running query.

Reply:
Ah, that makes sense.